services/121-service/test/registrations/pagination/filter-registration.test.ts

```diff
@@ -165,5 +165,28 @@ describe('Load PA table', () => {
       }
       expect(meta.totalItems).toBe(1);
     });
+
+    it('should filter using a referenceId string that could be interpreted as number', async () => {
+      // Arrange
+      const referenceIdInterpretableAsNumber = '651581942751358e5'; // A just-number string like '1234567890' is somehow not sufficiently covering the edge case, as this would not lead to the same bug prior to bugfix AB#30713
+      // Don't update existing registration, because updating referenceId is not preferable, and it messes up the other tests in this file
+      const newRegistration = structuredClone(registrationOCW1);
+      newRegistration.referenceId = referenceIdInterpretableAsNumber;
+      await importRegistrations(programIdOCW, [newRegistration], accessToken);
+
+      // Act
+      const getRegistrationsResponse = await getRegistrations({
+        programId: programIdOCW,
+        accessToken,
+        filter: {
+          'filter.referenceId': `$eq:${referenceIdInterpretableAsNumber}`,
```

Comment:
Maybe it wouild be nice to cover the other cases all well, for completeness (ie, `$in:`, `$ilike` etc.)